libs/datagrid/src/lib/filters/set-filter.tsx

```diff
@@ -9,85 +9,90 @@ import {
+        },
 
-      isFilterActive() {
-        return valueRef.current.length !== 0;
-      },
+        getModel() {
+          if (!this.isFilterActive()) {
+            return null;
+          }
+          return { value: valueRef.current };
+        },
 
-      getModel() {
-        if (!this.isFilterActive()) {
-          return null;
-        }
-        return { value: valueRef.current };
-      },
+        setModel(model?: { value: string[] } | null) {
+          valueRef.current = !model ? [] : model.value;
+          setValue(valueRef.current);
+        },
+      };
+    });
 
-      setModel(model?: { value: string[] } | null) {
-        valueRef.current = !model ? [] : model.value;
-        setValue(valueRef.current);
-      },
+    const onChange = (event: ChangeEvent<HTMLInputElement>) => {
+      valueRef.current = event.target.checked
+        ? [...value, event.target.value]
+        : value.filter((v) => v !== event.target.value);
+      setValue(valueRef.current);
     };
-  });
 
-  const onChange = (event: ChangeEvent<HTMLInputElement>) => {
-    valueRef.current = event.target.checked
-      ? [...value, event.target.value]
-      : value.filter((v) => v !== event.target.value);
-    setValue(valueRef.current);
-  };
-
-  useEffect(() => {
-    props.filterChangedCallback();
-  }, [value]); //eslint-disable-line react-hooks/exhaustive-deps
-
-  return (
-    <div className="ag-filter-body-wrapper">
-      <fieldset className="ag-simple-filter-body-wrapper">
-        {Object.keys(props.colDef.filterParams.set).map((key) => (
-          <label className="flex" key={key}>
-            <input
-              type="checkbox"
-              value={key}
-              className="mr-1"
-              checked={value.includes(key)}
-              onChange={onChange}
-            />
-            <span>{props.colDef.filterParams.set[key]}</span>
-          </label>
-        ))}
-      </fieldset>
-      <div className="ag-filter-apply-panel">
-        <button
-          type="button"
-          className="ag-standard-button ag-filter-apply-panel-button"
-          onClick={() => setValue((valueRef.current = []))}
-        >
-          {t('Reset')}
-        </button>
+    useEffect(() => {
+      props.filterChangedCallback();
+    }, [value]); //eslint-disable-line react-hooks/exhaustive-deps
+    return (
+      <div className="ag-filter-body-wrapper">
+        <fieldset className="ag-simple-filter-body-wrapper">
+          {Object.keys(props.colDef.filterParams.set).map((key) => (
+            <label className="flex" key={key}>
+              <input
+                type="checkbox"
+                value={key}
+                disabled={readonly}
+                className="mr-1"
+                checked={value.includes(key)}
+                onChange={onChange}
+              />
+              <span>{props.colDef.filterParams.set[key]}</span>
+            </label>
+          ))}
+        </fieldset>
+        {!readonly && (
```

Comment:
question (non blocking): why conditionally render the reset button? It doesn't seem necessary to me? If you are in view as mode, you'd probably still want to be able to filter

Reply:
If you are on active, closed, rejected tabs orders status filter value is preselected and filter is readonly. You can see what is selected but input's are disabled. input[type=checkbox] get's different styles so it's obvious for the user that he cannot change value.
If you cannot change value why we should render reset button? We can make it disabled, but it will not change styles. Of course we can change styles but it's just easier and less confusing just to hide it